api-gateway/src/main/java/com/maksimpegov/apigateway/config/AuthWebFilter.java

```diff
@@ -40,17 +39,25 @@ public Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
 			return response.setComplete();
 		}
 
-		// Modify an existing header
+		// If the header value is not a valid JWT, abort the request
 		String existingHeader = headers.getFirst(headerName);
 		if (!jwtService.isTokenValid(existingHeader)) {
 			response.setStatusCode(HttpStatus.UNAUTHORIZED);
 			return response.setComplete();
 		}
 
+		// Change JWT header value to user id
```

Comment:
Have you checked than UserId that you added here not returning to the customer with response?

Reply:
no, users service will receive user id,  customer will receive UNAUTHORIZED or users service response